GetStream/stream-chat-android
# [3468] collect events as soon as startListening gets invoked

### 🎯 Goal

Fix the Disconnected state after the app relaunches

### 🧪 Testing

- kill the app
- launch it again
- repeat 3-5 times

Make sure the app is not stuck in a Disconnected state.

### ☑️Contributor Checklist

#### General

#### Code & documentation

### ☑️Reviewer Checklist

### 🎉 GIF

_Please provide a suitable gif that describes your work on this pull request_
